src/kademlia/node.cpp

```diff
@@ -1210,111 +1062,58 @@ void node::incoming_request(msg const& m, entry& e)
 			VALGRIND_CHECK_MEM_IS_DEFINED(msg_keys[4].string_ptr(), item_sig_len);
 			VALGRIND_CHECK_MEM_IS_DEFINED(pk, item_pk_len);
 #endif
+			boost::uint64_t seq = boost::uint64_t(msg_keys[2].int_value());
```

Comment:
I think this is a dangerous cast. If someone announces a negative sequence number, you will interpret it as positive here (and presumably more recent than the authentic item). I would think it's safer to consistently use signed int64_t for sequence numbers